File: assets/api/AddSite.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Net;
using System.IO;
using System.Windows.Forms;

namespace APITest
{
    class SiteHelper
    {
        private void btnSite_Click(object sender, EventArgs e)
        {
            string url = "http://worldwater.byu.edu/app/index.php/default/services/api/sites";

            System.Net.HttpWebRequest request = (HttpWebRequest)System.Net.HttpWebRequest.Create(url);

            request.Method = "POST";

            request.ContentType = "application/json";

            using (var streamWriter = new StreamWriter(request.GetRequestStream()))
            {
                string json = @"{'user': 'admin',
                        'password': 'password',
                        'SourceID':20,
                        'SiteCode':'Te-15',
                        'SiteName':'test',
                        'Latitude':40.24182,
                        'Longitude':-111.61917,
                        'SiteType':'Stream',
                        'Elevation_m':1940.6,
                        'State':'Utah',
                        'County':'Utah',
                        'Comments':'citation001'}";

                json = json.Replace("'", "\"");
                streamWriter.Write(json);
                streamWriter.Flush();
                streamWriter.Close();
            }


            try
            {
                using (WebResponse response = request.GetResponse())
                {
                    using (var streamReader = new StreamReader(response.GetResponseStream()))
                    {
                        var result = streamReader.ReadToEnd();
                        MessageBox.Show(result);
                    }
                }
            }
            catch (WebException ex)
            {
                using (WebResponse response = ex.Response)
                {
                    HttpWebResponse httpResponse = (HttpWebResponse)response;
                    string errorCode = string.Format("Error code: {0} ", httpResponse.StatusCode);
                    using (Stream data = response.GetResponseStream())
                    using (var reader = new StreamReader(data))
                    {
                        string text = reader.ReadToEnd();
                        MessageBox.Show(errorCode + text);
                    }
                }
            }
        }
    }
}

File: assets/api/add_site.py
import urllib2
import json
import random

# this is a sample python client for the JSON API for adding a site
# author: Jiri Kadlec

username = 'admin'
password = 'password'

siteID = int(random.random() * 1000)
randomLat = random.random() * 180.0 - 90.0
randomLon = random.random() * 360.0 - 180.0
randomElev = random.random() * 1000
siteName = "Python Site %s" % siteID

#STEP 1: prepare the uploaded data in JSON format
data = {
    "user": username,
    "password": password,
    "SourceID": 35,
    "SiteName": siteName,
    "SiteCode": "python-%s" % siteID,
    "Latitude": randomLat,
    "Longitude": randomLon,
    "SiteType": "Atmosphere",
    "Elevation_m": randomElev,
    "Comments": "site uploaded from Python"
}
print data
postdata = json.dumps(data)


#STEP 2: post the data values to HydroServer and check response status
uploadURL = 'http://worldwater.byu.edu/app/index.php/default/services/api/sites'
req = urllib2.Request(uploadURL)
req.add_header('Content-Type', 'application/json')

try:
    response = urllib2.urlopen(req, postdata)
    print response.read()

except urllib2.HTTPError, e:
    print e.code
    print e.msg
    print e.headers
    print e.fp.read()
